site/app/controllers/OfficeHoursQueueController.php

```diff
@@ -32,31 +31,58 @@ public function showQueue() {
                 new RedirectResponse($this->core->buildCourseUrl(['home']))
             );
         }
-        if (!$this->core->getUser()->accessGrading()) {
-            $oh_queue = $this->core->getQueries()->getQueueByUser($this->core->getUser()->getId());
-            return Response::WebOnlyResponse(
-                new WebResponse(
-                    'OfficeHoursQueue',
-                    'showQueueStudent',
-                    $oh_queue,
-                    $this->core->getConfig()->getCourse()
-                )
+
+        return Response::WebOnlyResponse(
+            new WebResponse(
+                'OfficeHoursQueue',                      //Goes to this file OfficeHoursQueueView.php
+                'showTheQueue',                          //Runs this functin showTheQueue()
+                new OfficeHoursQueueViewer($this->core)  //Passing in this variable which is a OfficeHoursQueueViewer object
+            )
+        );
+    }
+
+    /**
+    * @Route("/{_semester}/{_course}/office_hours_queue/open", methods={"POST"})
+    * @AccessControl(role="LIMITED_ACCESS_GRADER")
+    * @return Response
+    */
+    public function openQueue() {
+        if (!isset($_POST['code']) || $_POST['code'] == "") {
+            $this->core->addErrorMessage("No code was provided");
+            return Response::RedirectOnlyResponse(
+                new RedirectResponse($this->core->buildCourseUrl(['office_hours_queue']))
             );
         }
-        else {
-            $oh_queue = $this->core->getQueries()->getInstructorQueue();
-            return Response::WebOnlyResponse(
-                new WebResponse(
-                    'OfficeHoursQueue',
-                    'showQueueInstructor',
-                    $oh_queue,
-                    $this->core->getConfig()->getCourse()
-                )
+
+        //Replace whitespace with "_"
+        $queue_code = preg_replace('/\s+/', '_', $_POST['code']);
+
+        $re = '/^[a-zA-Z0-9_\-]+$/m';
+        preg_match_all($re, $queue_code, $matches, PREG_SET_ORDER, 0);
+        if (count($matches) != 1) {
+            $this->core->addErrorMessage('Queue Code must only contain letters, numbers, spaces, "_", and "-"');
+            return Response::RedirectOnlyResponse(
+                new RedirectResponse($this->core->buildCourseUrl(['office_hours_queue']))
             );
         }
+
+        // Print the entire match result
+        echo count($matches) != 1;
```

Comment:
why is there an echo here?

Reply:
My bad that was from testing something. Good catch